src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,24 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(()
+                -> cache.computeIfAbsent(userId, this::fetchUserData), executor);
+    }
+
+    private UserData fetchUserData(String userId) {
+        return new UserData(userId, "Details for " + userId);
```

Comment:
we should add some delay in a form of `Thread.sleep(...)` following the requirements to simulate that some work is taking some time, like accessing the DB, and that if the data is already present in the cache map - we do not perform that task